File: sthlmtraveling/src/main/java/com/markupartist/sthlmtraveling/provider/TransportMode.java
package com.markupartist.sthlmtraveling.provider;

import android.text.TextUtils;

public class TransportMode {
    public final static int UNKNOWN_INDEX = -1;
    public final static int METRO_INDEX = 0;
    public final static int BUS_INDEX = 1;
    public final static int TRAIN_INDEX = 2;
    public final static int TRAM_INDEX = 3;
    public final static int BOAT_INDEX = 4;
    public final static int FOOT_INDEX = 5;
    public final static int NAR_INDEX = 6;

    public static final String UNKNOWN = "";
    public static final String METRO = "MET";
    public static final String METRO_SYNONYM = "METRO";
    public static final String BUS = "BUS";
    public static final String TRAIN = "TRN";
    public static final String TRAM = "TRM";
    public static final String BOAT = "SHP";
    public static final String WAX = "WAX"; // Used in the request.
    public static final String FOOT = "Walk";
    public static final String NAR = "NAR";

    public static final String FLY = "FLY";
    public static final String AEX = "AEX";

    public static int getIndex(String transportMode) {
        if (TextUtils.isEmpty(transportMode)) {
            return UNKNOWN_INDEX;
        }
        switch (transportMode) {
            case METRO:
            case METRO_SYNONYM:
                return METRO_INDEX;
            case BUS:
                return BUS_INDEX;
            case TRAIN:
                return TRAIN_INDEX;
            case TRAM:
                return TRAM_INDEX;
            case FOOT:
                return FOOT_INDEX;
            case BOAT:
                return BOAT_INDEX;
            case NAR:
                return NAR_INDEX;
        }
        return UNKNOWN_INDEX;
    }
}

File: sthlmtraveling/src/main/java/com/markupartist/sthlmtraveling/provider/TransportMode.kt
package com.markupartist.sthlmtraveling.provider

import android.text.TextUtils

object TransportMode {
    const val UNKNOWN_INDEX: Int = -1
    const val METRO_INDEX: Int = 0
    const val BUS_INDEX: Int = 1
    const val TRAIN_INDEX: Int = 2
    const val TRAM_INDEX: Int = 3
    const val BOAT_INDEX: Int = 4
    const val FOOT_INDEX: Int = 5
    const val NAR_INDEX: Int = 6

    const val UNKNOWN: String = ""
    const val METRO: String = "MET"
    const val METRO_SYNONYM: String = "METRO"
    const val BUS: String = "BUS"
    const val TRAIN: String = "TRN"
    const val TRAM: String = "TRM"
    const val BOAT: String = "SHP"
    const val WAX: String = "WAX" // Used in the request.
    const val FOOT: String = "Walk"
    const val NAR: String = "NAR"

    const val FLY: String = "FLY"
    const val AEX: String = "AEX"
    const val BIKE_RENTAL: String = "bikeRental"

    @JvmStatic
    fun getIndex(transportMode: String?): Int {
        if (TextUtils.isEmpty(transportMode)) {
            return UNKNOWN_INDEX
        }
        when (transportMode) {
            METRO, METRO_SYNONYM -> return METRO_INDEX
            BUS -> return BUS_INDEX
            TRAIN -> return TRAIN_INDEX
            TRAM -> return TRAM_INDEX
            FOOT -> return FOOT_INDEX
            BOAT -> return BOAT_INDEX
            NAR -> return NAR_INDEX
        }
        return UNKNOWN_INDEX
    }
}
